lib/logstash/filters/dns.rb

```diff
@@ -219,8 +222,9 @@ def resolve(event)
         end
       rescue Resolv::ResolvTimeout
         @failed_cache[raw] = true if @failed_cache
-        @logger.warn("DNS: timeout on resolving the hostname.",
+        @logger.info("DNS: timeout on resolving the hostname.",
```

Comment:
While we have a _request_ to change the logging level, it seems to me that this may break in-the-wild configs that _rely_ on the current logging level. Can/should we make this behaviour configurable?

Reply:
additionally this change is not covered in the specs.

Reply:
Added `warn_on_timeout` flag to provide flexibility. Also covered the unit test cases.